core/commands/config.go

```diff
@@ -338,12 +346,39 @@ var configProfileApplyCmd = &cmds.Command{
 			return
 		}
 
-		err := transformConfig(req.InvocContext().ConfigRoot, req.Arguments()[0], profile.Transform)
+		dryRun, _, _ := req.Option("dry-run").Bool()
+		diff, err := transformConfig(req.InvocContext().ConfigRoot, req.Arguments()[0], profile.Transform, dryRun)
 		if err != nil {
 			res.SetError(err, cmdkit.ErrNormal)
 			return
 		}
-		res.SetOutput(nil)
+		res.SetOutput(&ConfigProfileApplyOutput{
+			Diff: *diff,
+		})
+	},
+	Marshalers: cmds.MarshalerMap{
+		cmds.Text: func(res cmds.Response) (io.Reader, error) {
+			v, err := unwrapOutput(res.Output())
+			if err != nil {
+				return nil, err
+			}
+
+			apply, ok := v.(*ConfigProfileApplyOutput)
+			if !ok {
+				return nil, e.TypeErr(apply, v)
+			}
+
+			dryRun, _, err := res.Request().Option("dry-run").Bool()
```

Comment:
We actually realized that we'd prefer to *always* print out the diff, even if we're *not* doing a dry-run (https://github.com/ipfs/go-ipfs/pull/4923#discussion_r182314868). We can just return `strings.NewReader(jsondiff.Format(apply.Diff))` without checking `dryRun`.